src/Toml/Codec/Combinator/Custom.hs

```diff
@@ -235,3 +241,18 @@ validateIf p biMap k = validate validateEither biMap k
         if p a
         then Right a
         else Left $ "Value does not pass the validation for key: " <> prettyKey k
+
+{- | Converts a given @TomlCodec@ to give you access to the list of
+@TomlDecodeError@s as an @Either@
+
+__Example:__
+
+@
+either (bool "mykey") :: TomlCodec (Either [TomlDecodeError] Bool)
+@
+-}
+either :: TomlCodec a -> TomlCodec (Either [TomlDecodeError] a)
+either (Codec r w) = Codec r' $ \case
+  Left x' -> pure . Left $ x'
+  Right x' -> fmap Right . w $ x'
```

Comment:
This case looks like the `Traversable` instance for `Either`, so you can use a single function `traverse` instead of a manual case:

* https://hackage.haskell.org/package/base-4.14.0.0/docs/src/Data.Traversable.html#line-240

Reply:
You're totally right! :wink: